frame/support/src/storage/mod.rs

```diff
@@ -30,6 +30,83 @@ pub mod child;
 pub mod generator;
 pub mod migration;
 
+#[cfg(not(build_type = "release"))]
+mod debug_helper {
+	#[cfg(feature = "std")]
+	mod with_std {
+		use sp_std::cell::RefCell;
+
+		thread_local! {
+			static TRANSACTION_LEVEL: RefCell<u32> = RefCell::new(0);
+		}
+
+		pub fn require_transaction() {
+			let level = TRANSACTION_LEVEL.with(|v| *v.borrow());
+			if level == 0 {
+				panic!("Require transaction not called within with_transaction");
+			}
+		}
+
+		pub fn inc_transaction_level() {
+			TRANSACTION_LEVEL.with(|v| {
+				let mut val = v.borrow_mut();
+				*val += 1;
+				if *val > 10 {
+					crate::debug::warn!("Detected with_transaction with nest level {}. Nested usage of with_transaction is not recommended.", *val);
+				}
+			});
+		}
+
+		pub fn dec_transaction_level() {
+			TRANSACTION_LEVEL.with(|v| *v.borrow_mut() -= 1);
+		}
+	}
+
+	#[cfg(feature = "std")]
+	pub use with_std::*;
+
+	#[cfg(not(feature = "std"))]
+	mod without_std {
```

Comment:
Do we really need this in WASM?

Reply:
It maybe useful to be able to run this in testnet? Not too sure, can remove if people think we don't need this in wasm.

Reply:
Removed for now. Can add it back later if someone actually wants it.